src/Std/Data/DHashMap/Lemmas.lean

```diff
@@ -964,64 +964,666 @@ theorem distinct_keys [EquivBEq α] [LawfulHashable α] :
+    if heq : k == k' then
+      haveI : Inhabited (β k) := by rwa [eq_of_beq heq]
+      -- not correct if f does not preserve default: ... f (m.get! k)
+      -- possible alternative: write ... (m.getD k (cast ⋯ default))
+      m.get? k |>.map f |>.map (cast (congrArg β (eq_of_beq heq))) |>.get!
+    else
+      m.get! k' :=
+  sorry
+
+@[simp]
+theorem get!_modify_self [LawfulBEq α] {k : α} [Inhabited (β k)] {f : β k → β k} :
+    (m.modify k f).get! k = ((m.get? k).map f).get! :=
+  sorry
+
+theorem getD_modify [LawfulBEq α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getD k' =
+    if heq : k == k' then
+      m.get? k |>.map f |>.map (cast (congrArg β <| eq_of_beq heq)) |>.getD
+    else
+      m.getD k' :=
+  sorry
+
+@[simp]
+theorem getD_modify_self [LawfulBEq α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getD k = ((m.get? k).map f).getD :=
+  sorry
+
+theorem getKey?_modify [LawfulBEq α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getKey? k' =
+    if k == k' then
+      Option.guard (fun _ => k ∈ m) k
+    else
+      m.getKey? k' :=
+  sorry
+
+@[simp]
+theorem getKey?_modify_self [LawfulBEq α] {k : α} {f : β k → β k} :
+    (m.modify k f).getKey? k = Option.guard (fun _ => k ∈ m) k :=
+  sorry
+
+theorem getKey!_modify [LawfulBEq α] [Inhabited α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getKey! k' =
+    if k == k' then
+      -- possible alternative (here and at other places): if k ∈ m then k else default
+      -- although correct, I'm not sure about the tradeoffs of replacing get! with
+      -- something that does not panic
+      Option.guard (fun _ => k ∈ m) k |>.get!
+    else
+      m.getKey! k' :=
+  sorry
+
+@[simp]
+theorem getKey!_modify_self [LawfulBEq α] [Inhabited α] {k : α} {f : β k → β k} :
+    (m.modify k f).getKey! k = (Option.guard (fun _ => k ∈ m) k).get! :=
+  sorry
+
+theorem getKey_modify [LawfulBEq α] [Inhabited α] {k k' : α} {f : β k → β k}
+    (h : k' ∈ m.modify k f) :
+    (m.modify k f).getKey k' h =
+    if heq : k == k' then
+      k
+    else
+      haveI h' : k' ∈ m := by rwa [mem_modify] at h
+      m.getKey k' h' :=
+  sorry
+
+@[simp]
+theorem getKey_modify_self [LawfulBEq α] [Inhabited α] {k : α} {f : β k → β k}
+    (h : k ∈ m.modify k f) : (m.modify k f).getKey k h = k :=
+  sorry
+
+theorem getKeyD_modify [LawfulBEq α] {k k' : α} {f : β k → β k} :
+    (m.modify k f).getKeyD k' =
+    if k == k' then
+      if k ∈ m then (fun _ => k) else (·)
+    else
+      m.getKeyD k' :=
+  sorry
+
+-- if is sufficiently simple that simp might be okay
+@[simp]
+theorem getKeyD_modify_self [LawfulBEq α] [Inhabited α] {k : α} {f : β k → β k} :
+    (m.modify k f).getKeyD k = if k ∈ m then (fun _ => k) else (·) :=
+  sorry
+
+theorem modify_eq_alter [LawfulBEq α] {k : α} {f : β k → β k} :
+    m.modify k f = m.alter k (·.map f) := sorry
+
+variable {β : Type v} {m : DHashMap α (fun _ => β)} in
+theorem modify_eq_constAlter [LawfulBEq α] {k : α} {f : β → β} :
```

Comment:
This would usually be `modify_eq_const_alter`: namespace dots turn into snake case, not camel case, in theorem names.

Reply:
Is this true? I used to think that, but after seeing changes like https://github.com/leanprover-community/mathlib4/pull/18808 I no longer thought it worked like this. Or does that one only get to be camel-cased because it's a notation?

Reply:
Indeed, I'm out of date here, and indeed #naming is explicit on this question:
```quote
If the definition name is unambiguous without its namespace, it can be used as is. Else, the namespace is prepended back to it in lowerCamelCase. This is to ensure that _-separated strings in a lemma name correspond to a definition name or connective.
```